allensdk/brain_observatory/behavior/behavior_project_cache/tables/util/metadata_parsers.py

```diff
@@ -0,0 +1,245 @@
+    "OPHYS_5_images_A_passive",
+    "OPHYS_5_images_B_passive",
+    "OPHYS_5_images_H_passive",
+    "OPHYS_6_images_A",
+    "OPHYS_6_images_B",
+    "OPHYS_6_images_H",
+    "TRAINING_0_gratings_autorewards_15min",
+    "TRAINING_1_gratings",
+    "TRAINING_2_gratings_flashed",
+    "TRAINING_3_images_A_10uL_reward",
+    "TRAINING_3_images_B_10uL_reward",
+    "TRAINING_3_images_G_10uL_reward",
+    "TRAINING_4_images_A_handoff_lapsed",
+    "TRAINING_4_images_A_handoff_ready",
+    "TRAINING_4_images_A_training",
+    "TRAINING_4_images_B_training",
+    "TRAINING_4_images_G_training",
+    "TRAINING_5_images_A_epilogue",
+    "TRAINING_5_images_A_handoff_lapsed",
+    "TRAINING_5_images_A_handoff_ready",
+    "TRAINING_5_images_B_epilogue",
+    "TRAINING_5_images_B_handoff_lapsed",
+    "TRAINING_5_images_B_handoff_ready",
+    "TRAINING_5_images_G_epilogue",
+    "TRAINING_5_images_G_handoff_lapsed",
+    "TRAINING_5_images_G_handoff_ready",
+]
+
+PROJECT_CODES = [
+    "VisualBehavior",
+    "VisualBehaviorTask1B",
+    "VisualBehaviorMultiscope",
+    "VisualBehaviorMultiscope4areasx2d",
+]
+
+NUM_STRUCTURES_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 2,
+    "VisualBehaviorMultiscope4areasx2d": 4,
+}
+
+NUM_DEPTHS_DICT = {
+    "VisualBehavior": 1,
+    "VisualBehaviorTask1B": 1,
+    "VisualBehaviorMultiscope": 4,
+    "VisualBehaviorMultiscope4areasx2d": 2,
+}
+
+OPHYS_BEHAV_TYPE_DICT = {
+    "active_behavior": ["0", "1", "3", "4", "6"],
+    "passive_viewing": ["2", "5", "7"],
+}
+
+
+############################
+#
+#     from session type
+#
+############################
+
+
+def parse_behavior_context(metadata_row: pd.Series) -> Union[str, None]:
+    """Return if session is passive or active.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    behavior_type : str
+        Return string describing active or passive session.
+    """
+    passive_ophys_session_nums = ["2", "5", "7"]
+
+    session_type = metadata_row["session_type"]
+
+    if session_type not in SESSION_TYPES:
+        return None
+    is_passive = (
+        any(
+            passive_num in session_type
+            for passive_num in passive_ophys_session_nums
+        )
+        and "OPHYS" in session_type
+    )
+    if is_passive:
+        behavior_type = "passive_viewing"
+    else:
+        behavior_type = "active_behavior"
+    return behavior_type
+
+
+def parse_stimulus_set(metadata_row: pd.Series) -> Union[str, None]:
+    """Return the name of the image set or gratings.
+
+    Parameters
+    ----------
+    metadata_row : pandas.Series
+        Complete set of metadata for session.
+
+    Returns
+    -------
+    stim_set : str
+        Name of stimulus set shown for session.
+    """
+    session_type = metadata_row["session_type"]
+    if session_type not in SESSION_TYPES:
```

Comment:
I don't like silently returning `None` on error. Also seems like you don't need to check this. Why not just raise if we are unable to parse?

Reply:
I was thinking of this as filling values in a pandas dataframe. Much like other missing data where we fill with `NaN` or `<N/A>`, I was treating this similarly.